modules/core/src/main/java/org/apache/synapse/message/processor/impl/forwarder/ForwardingService.java

```diff
@@ -358,7 +358,7 @@ public void init(SynapseEnvironment se) {
         if (nonRetryStatusCodes != null) {
             for (String code : nonRetryStatusCodes) {
                 try {
-                    int codeI = Integer.parseInt(code);
+                    int codeI = Integer.parseInt(code.trim());
```

Comment:
This fix is not the part of the addressed issue, but since there is a possibility to enter the comma separate values with spaces [ eg: 304, 305]  making this improvement